simulators/portal-interop/src/main.rs

```diff
@@ -641,33 +641,53 @@ dyn_async! {
             }
         }
 
-        // wait 8 seconds for data to propagate
-        // This value is determined by how long the sleeps are in the gossip code
-        // So we may lower this or remove it depending on what we find.
-        tokio::time::sleep(Duration::from_secs(8)).await;
-
-        let comments = vec!["1 header", "1 block body", "100 header",
-            "100 block body", "7000000 header", "7000000 block body",
-            "7000000 receipt", "15600000 (post-merge) header", "15600000 (post-merge) block body", "15600000 (post-merge) receipt",
-            "17510000 (post-shanghai) header", "17510000 (post-shanghai) block body", "17510000 (post-shanghai) receipt"];
-
+        // wait content_vec.len() seconds for data to propagate, giving more time if more items are propagating
+        tokio::time::sleep(Duration::from_secs(test_data.len() as u64)).await;
+
+        let (first_header_key, first_header_value) = test_data.get(0).unwrap();
```

Comment:
Could you elaborate on what is happening here, what the first header and last header are, why we set the first as last, and how this relates to comments?

Reply:
Curious about this too.

Reply:
Last header seen. We are setting it here because the python borrow checker won't allow us to set unassigned variables.

So at the start the last seen header would be the first header, because it is the only one we know about at the start therefore being the last header seen

Reply:
It's worth adding a docstring to the code for reference